sktime/annotation/hmm.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+"""
+HMM Annotation Estimator.
+
+Implements a basic Hidden Markov Model (HMM) as an annotation estimator.
+To read more about the algorithm, check out the wikipedia page:
+(https://en.wikipedia.org/wiki/Hidden_Markov_model
+"""
+from math import isclose
+from typing import Tuple
+
+import numpy as np
+
+from sktime.annotation.base._base import BaseSeriesAnnotator
+
+__author__ = ["miraep8"]
+__all__ = ["HMM"]
+
+
+class HMM(BaseSeriesAnnotator):
+    """Implements a simple HMM fitted with viterbi algorithm.
+
+    The HMM annotation estimator facilitates the appilcation of
+    the vertibi algorithm to fit a sequence of 'hidden state' class
+    annotations (represented by an array of integers the same size
+    as the observation) to a sequence of observations.
+
+    This is done by finding the most likely path given the emission
+    probabilities - (ie the probability that a particular observation
+    would be generated by a given hidden state), the transition prob
+    (ie the probability of transitioning from one state to another or
+    staying in the same state) and the initial probabilities - ie the
+    belief of the probability distribution of hidden states at the
+    start of the observation sequence).
+
+    Parameters
+    ----------
+    emission_funcs : a list of functions, which should be properly
+        normalized PDFs over the same space as the observed data.
+    transition_prob_mat: a nxn array of probabilities, where 'n'
+        represents the number of hidden states in the HMM.  Each
+        row should sumn to 1 in order to be properly normalized
+        (ie the j'th column in the i'th row represents the
+        probability of transitioning from state i to state j.)
+    initial_probs: optional, a 1d array of probabilities across
+        the starting states. Should match prior beliefs.  If none
+        is passed will give each state an equal initial probability.
+
+    Examples
+    --------
+    >>> from sktime.annotation.hmm import HMM
+    >>> from scipy.stats import norm
+    >>> from numpy import asarray
+    >>> # define the emission probs for our HMM model:
+    >>> centers = [3.5,-5]
+    >>> sd = [.25 for i in centers]
+    >>> emi_funcs = [(norm.pdf, {'loc': mean,
+    ...  'scale': sd[ind]}) for ind, mean in enumerate(centers)]
+    >>> test = HMM(emi_funcs, asarray([[0.25,0.75], [0.666, 0.333]]))
```

Comment:
re naming, avoid calling things `test` or `temp` or similar. Names, even if short, should be as descriptive as possible. Better, perhaps: `hmm`, `hmm_est`